[Feature] Enable `kwargs` for single component models (via `extra`) (#1032)
mckinsey/vizro: vizro-core/docs/pages/user-guides/container.md

This guide shows you how to use containers to group your components into sections and subsections within the page.

A [Container][vizro.models.Container] complements a [Page][vizro.models.Page], and both models share nearly identical arguments. While `Page.layout` provides a method for structuring the overall page layout, a `Container` offers more detailed control within a particular section of the page.

> Unlike the `Page.layout`, the `Container` includes a `background` argument, allowing you to color its background to differentiate it from the rest of the page. Additional functionality will soon be added to the Container, including controls specific to it, which will further enhance the management of related components.

Reviewer: ```suggestion
Unlike the `Page.layout`, the `Container` includes a `background` argument, enabling you to color its background to differentiate it from the rest of the page. Additional functionality will soon be added to the `Container`, including controls specific to it, which will further enhance the management of related components.
```

To create nested containers, add a `Container` to the `components` argument of another `Container`.

```python title="Example"
vm.Container(
    title="Parent Container",
    components=[
        vm.Container(
            title="Child Container",
            components=[vm.Button()],
        )
    ],
)
```

## The `extra` argument

Currently the `Container` is based on the underlying [`dbc.Container`](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/layout/). Using the `extra` argument you pass additional arguments to the underlying object in order to alter it beyond the chosen defaults.

!!! warning
    Using `extra` is a quick and flexible way to alter a component beyond what Vizro offers. However, [it is not a part of the official Vizro schema](../explanation/schema.md#what-is-the-vizro-json-schema) and we do not consider it a breaking change if we alter/remove it. This is unlikely to happen any time soon.

> An example use would be to enhance the visibility of the `Container` as a distinct section within your dashboard. For this, you can use `extra={"class_name": "bg-container"}`.

Reviewer 2: This will be built in once #1002 is merged so we'll need a different example.

@huong-li-nguyen any ideas for a nice example here? What's something you might like to do to `Container` and can't achieve with `theme` alone?

Reviewer 3: https://dash-bootstrap-components.opensource.faculty.ai/docs/components/layout/

Nothing really that is suited for a visual example. `tag` and `fluid` are things people might want to configure, but visually this changes nothing. But do we even need an example of this extra argument for every component? 😅

Author: They felt forced to me as well, but then again it's quite hard to do on your own if you have never seen it done. I already omitted the examples for all the form components, so would somewhat like to keep at least the ones for `Container`, `Card` and `Button`..

Between us, I feel more and more that with the `style` argument, why do we even have `extra` anymore 😂 , but that's a whole other can to open now!

If you want to style your `Container` beyond the styling options available inside `variant`, please refer to our user guide on [overwriting CSS for selected components](custom-css.md#overwrite-css-for-selected-components).

## The `extra` argument

Currently the `Container` is based on the underlying [`dbc.Container`](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/layout/). Using the `extra` argument you can pass additional arguments to the underlying object in order to alter it beyond the chosen defaults.

!!! warning
    Using `extra` is a quick and flexible way to alter a component beyond what Vizro offers. However, [it is not a part of the official Vizro schema](../explanation/schema.md#what-is-the-vizro-json-schema) and we do not consider it a breaking change if we alter/remove it. This is unlikely to happen any time soon.

> For examples of how to use the `extra` argument, we provided a comprehensive example in the documentation of [`Card`](card-button.md#the-extra-argument).

Reviewer 2: ```suggestion
For examples of how to use the `extra` argument, we provide an example in the documentation of [`Card`](card-button.md#the-extra-argument).
```
